docs/rfds/session-config-options.mdx

```diff
@@ -0,0 +1,240 @@
+
+The Client can render the options provided, send updated values to the Agent, and also display any changes the Agent made during the course of it's execution (for example, if it changes modes or models because of fallbacks or a change in strategy, so that the user can always see the current state).
+
+Since we are moving to a world in which there are multiple configuration options, some of which may depend on each other, the Agent MUST provide the complete set of configuration options and their current values whenever a change is made. We would tradeoff some extra data being sent to the Client in order to help minimize the amount of state required to be managed by the Client. The Client would submit a new value, and receive back the full state of all configuration options that it can then replace it's current state with and render. So if changing a model means there are no thinking options, or a new option becomes available, or another value needs to change because the values of an option are different, the Agent will reflect this in its response by providing the entire new state (or an error if it is somehow an invalid selection).
+
+## Implementation details and plan
+
+> Tell me more about your implementation. What is your detailed implementation plan?
+
+To start, we can implement this based on the [Session Modes](https://agentclientprotocol.com/protocol/session-modes) api.
+
+Something like an `InitializeResponse` that looks like:
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 1,
+  "result": {
+    "sessionId": "sess_abc123def456",
+    "config_options": [
+      {
+        "id": "mode", // this is the unique `key` for communication about which option is being used
+        "name": "Session Mode", // Human-readable label for the option
+        "description": "Optional description for the Client to display to the user."
+        "currentId": "ask",
+        "options": [
+          {
+            "id": "ask",
+            "name": "Ask",
+            "description": "Request permission before making any changes"
+          },
+          {
+            "id": "code",
+            "name": "Code",
+            "description": "Write and modify code with full tool access"
+          }
+        ]
+      },
+      {
+        "id": "models",
+        "name": "Model",
+        "currentId": "ask",
+        "options": [
+          {
+            "id": "model-1",
+            "name": "Model 1",
+            "description": "The fastest model"
+          },
+          {
+            "id": "model-2",
+            "name": "Model 2",
+            "description": "The most powerful model"
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+When we introduce this, we could also allow for grouped options, in case there are logical sub-headers and groupings for the options in an individual selector. It
```

Comment:
```suggestion
When we introduce this, we could also allow for grouped options, in case there are logical sub-headers and groupings for the options in an individual selector.
```